src/SDK/MonitoringApplianceEmulator/ViewModels/MetricChartPropertyControlViewModel.cs

```diff
@@ -0,0 +1,213 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="MetricChartPropertyControlViewModel.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.ViewModels
+{
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Globalization;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Windows;
+    using System.Windows.Media;
+    using LiveCharts;
+    using LiveCharts.Configurations;
+    using LiveCharts.Defaults;
+    using LiveCharts.Wpf;
+    using Microsoft.Azure.Monitoring.SmartDetectors.Metric;
+    using Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.Controls;
+    using Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.Models;
+    using Microsoft.Azure.Monitoring.SmartDetectors.RuntimeEnvironment.Contracts.AlertProperties;
+
+    /// <summary>
+    /// The view model class for the <see cref="MetricChartPropertyControl"/> control.
+    /// </summary>
+    public class MetricChartPropertyControlViewModel : ObservableObject
+    {
+        private static readonly Brush SeriesColor = Brushes.DeepSkyBlue;
+
+        private readonly ObservableTask<ChartValues<DateTimePoint>> loadChartTask;
+        private readonly IAnalysisServicesFactory analysisServicesFactory;
+        private readonly ITracer tracer;
+        private SeriesCollection seriesCollection;
+        private Func<double, string> xAxisFormatter;
+        private Func<double, string> yAxisFormatter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetricChartPropertyControlViewModel"/> class.
+        /// </summary>
+        /// <param name="analysisServicesFactory">The analysis services factory</param>
+        /// <param name="metricChartAlertProperty">The metric chart alert property that should be displayed.</param>
+        /// <param name="tracer">The tracer</param>
+        public MetricChartPropertyControlViewModel(
+            MetricChartAlertProperty metricChartAlertProperty,
+            IAnalysisServicesFactory analysisServicesFactory,
+            ITracer tracer)
+        {
+            this.Title = metricChartAlertProperty.DisplayName;
+            this.analysisServicesFactory = analysisServicesFactory;
+            this.tracer = tracer;
+
+            // Start a task to read the metric values
+            this.loadChartTask = new ObservableTask<ChartValues<DateTimePoint>>(
+                this.ReadChartValues(metricChartAlertProperty),
+                tracer,
+                this.DisplayChartValues);
```

Comment:
>this.DisplayChartValues [](start = 16, length = 23)

this is not the right pattern - the ObservableTask should return the data you bind to in the view
See for example SmartDetectorConfigurationControlViewModel.ReadResourcesTask